src/components/pagination/pagination.vue

```diff
@@ -1,96 +1,93 @@
-        direction="left"
-        :disabled="pageNumber === 1"
-      />
-      <paginationButton
-        :class="{ 'pagination__button--current': pageNumber === count }"
-        v-for="count in pageCount"
-        :number="count"
-        :key="count"
-        @paginationButtonClick="changePageNumber($event)"
-      />
-      <paginationArrow
-        @paginationArrowClick="changePageNumber($event)"
-        direction="right"
-        :disabled="pageNumber === pageCount"
-      />
-    </div>
+  <div class="pagination-wrapper">
+    <VPagination
+      v-model="page"
+      :pages="pages"
+      :range-size="rangeSize"
+      :active-color="activeColor"
+      :hideFirstButton="hideFirstButton"
+      :hideLastButton="hideLastButton"
+    />
   </div>
 </template>
 
 <script lang="ts">
-import { defineComponent, ref, computed } from 'vue'
-import paginationButton from '@/components/pagination/paginationButton.vue'
-import paginationArrow from '@/components/pagination/paginationArrow.vue'
-
-type changePageNumberEvent = {
-  number?: number
-  direction?: string
-}
+import { computed, defineComponent } from 'vue'
+import '@hennge/vue3-pagination/dist/vue3-pagination.css'
+import VPagination from '@hennge/vue3-pagination'
 
 export default defineComponent({
-  emits: ['changePageNumber'],
+  name: 'pagination',
+  components: { VPagination },
+  emits: ['update:modelValue'],
   props: {
-    totalLength: { type: [String, Number], default: 0 },
-    blocksPerPage: { type: Number, required: false, default: 3 },
+    modelValue: { type: Number, required: true },
+    pages: { type: Number, required: true },
+    rangeSize: { type: Number, default: 1 },
+    activeColor: { type: String, default: '#007bff' },
+    hideFirstButton: { type: Boolean, default: true },
+    hideLastButton: { type: Boolean, default: true },
   },
-  components: { paginationButton, paginationArrow },
   setup(props, { emit }) {
-    let pageNumber = ref<number>(1)
-    const pageCount = computed(() =>
-      Math.ceil((props.totalLength as number) / props.blocksPerPage)
-    )
-
-    const changePageNumber = (event: changePageNumberEvent) => {
-      pageNumber.value = event.number
-        ? event.number
-        : event.direction === 'right'
-        ? pageNumber.value + 1
-        : pageNumber.value - 1
+    const page = computed({
+      get: () => props['modelValue'],
+      set: (value) => emit(`update:modelValue`, value),
+    })
 
-      emit('changePageNumber', pageNumber.value)
-    }
-
-    return { pageCount, pageNumber, changePageNumber }
+    return { page }
   },
 })
 </script>
 
-<style scoped lang="scss">
-.pagination {
-  margin: 4rem 0 6rem 0;
-  &__buttons {
+<style lang="scss">
+.pagination-wrapper {
+  display: flex;
+  justify-content: center;
+
+  li {
+    background: #fff;
+    border: 0.1rem solid var(--clr__action);
+    border-radius: 0.4rem;
+    margin: 0 0.4rem;
+    min-width: 2.6rem;
+    height: 3.6rem;
+  }
+
+  button {
```

Comment:
class

Reply:
@ hennge / vue3-pagination package is used, this is the only way I can interact with elements of this pagination